modules/fedimint-ln-client/src/pay.rs

```diff
@@ -0,0 +1,545 @@
+            }
+            LightningPayStates::Refundable(refundable) => {
+                refundable.transitions(self.common.clone(), global_context.clone())
+            }
+            LightningPayStates::Refund(refund) => refund.transitions(&self.common, global_context),
+            LightningPayStates::Refunded(_) => {
+                vec![]
+            }
+            LightningPayStates::Failure(_) => {
+                vec![]
+            }
+        }
+    }
+
+    fn operation_id(&self) -> OperationId {
+        self.common.operation_id
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
+pub struct LightningPayCreatedOutgoingLnContract {
+    pub funding_txid: TransactionId,
+    pub contract_id: ContractId,
+    pub gateway: LightningGateway,
+}
+
+impl LightningPayCreatedOutgoingLnContract {
+    fn transitions(
+        &self,
+        common: &LightningPayCommon,
+        global_context: &DynGlobalClientContext,
+    ) -> Vec<StateTransition<LightningPayStateMachine>> {
+        let txid = self.funding_txid;
+        let contract_id = self.contract_id;
+        let funded_common = common.clone();
+        let rejected_common = funded_common.clone();
+        let success_context = global_context.clone();
+        let reject_context = global_context.clone();
+        let gateway = self.gateway.clone();
+        vec![
+            // Success case: funding transaction is accepted
+            StateTransition::new(
+                Self::await_outgoing_contract_funded(funded_common.clone(), success_context, txid),
+                move |_dbtx, (), old_state| {
+                    Box::pin(Self::transition_outgoing_contract_funded(
+                        old_state,
+                        funded_common.clone(),
+                        contract_id,
+                        gateway.clone(),
+                    ))
+                },
+            ),
+            // Failure case: funding transaction is rejected
+            StateTransition::new(
+                Self::await_funding_transaction_rejected(
+                    rejected_common,
+                    reject_context,
+                    self.funding_txid,
+                ),
+                |_dbtx, (), old_state| Box::pin(Self::transition_canceled(old_state)),
+            ),
+        ]
+    }
+
+    async fn await_outgoing_contract_funded(
+        common: LightningPayCommon,
+        global_context: DynGlobalClientContext,
+        txid: TransactionId,
+    ) {
+        global_context
+            .await_tx_accepted(common.operation_id, txid)
+            .await;
+    }
+
+    async fn transition_outgoing_contract_funded(
+        old_state: LightningPayStateMachine,
+        common: LightningPayCommon,
+        contract_id: ContractId,
+        gateway: LightningGateway,
+    ) -> LightningPayStateMachine {
+        assert!(matches!(
+            old_state.state,
+            LightningPayStates::CreatedOutgoingLnContract(_)
+        ));
```

Comment:
If you were using the data from inside `old_state.state` you wouldn't need to pass in the gateway, potentially saving some clones.